index.bs

```diff
@@ -0,0 +1,221 @@
+    1. For each <var>element</var> of <var>elements</var>:
+        1. Set <a>currentFrameStartingPoint</a> to the <a>starting point</a> of <var>element</var>.
+        1. Set <a>currentVisualRepresentation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>currentFrameStartingPoint</a> and <a>previousFrameStartingPoint</a>.  
+        1. If <var>unstable</var> is true, add a tuple to <var>unstable elements</var> with:
+            1. <a>currentVisualRepresentation</a> of <var>element</var>
+            1. <a>previousVisualRepresentation</a> of <var>element</var>
+            1. <var ignore=''>shiftFraction</var> of <var>element</var>:
+                1. Set to 0 if <a>previousFrameStartingPoint</a> is null.
+                1. Set to the max(abs(<a>previousFrameStartingPoint</a>.height - <a>currentFrameStartingPoint</a>.height) / <var>total height</var> , abs(<a>previousFrameStartingPoint</a>.width - <a>currentFrameStartingPoint</a>.width) / <var>total width</var>).
+        1. Set <a>previousFrameStartingPoint</a> point to <a>currentFrameStartingPoint</a>.
+        1. Set <a>previousVisualRepresentation</a> to the <a>currentVisualRepresentation</a>.
+    1. Call <a>compute the layout shift</a> with <var>unstable elements</var>.
+</div>
+
+Compute the Layout Shift {#sec-compute-layout-shift}
+-----------------------------------------------------
+
+<div algorithm="compute the layout shift">
+    When asked to to <dfn export>compute the layout shift</dfn>, with <var>unstable elements</var>, a list of impacted <a>element</a>'s on the page, as input, run the following steps:
+
+    1. Let <var>impacted region</var> be initially set to empty.
+    1. Let <var>max shift fraction</var> be initially set to 0.
+    1. For each tuple <var>tuple</var> in <var>unstable elements</var>:
+        1. Take the geometric union of <var>impacted region</var> with <var>tuple</var>’s <a>currentVisualRepresentation</a> and <var>tuple</var>’s <a>previousVisualRepresentation</a> if <a>previousVisualRepresentation</a> is not null.
+        1. If <var>tuple</var>’s <var>shiftFraction</var> > <var>max shift fraction</var>, <var>max shift fraction</var> equals <var>shiftFraction</var>.
+    1. Let <var>impacted fraction</var> be the area of the <var>impacted region</var> divided by the area of the current <a>viewport</a>.
+    1. The <var>layout shift</var> is the <var>impacted fraction</var> multiplied by the <var>max shift fraction</var>.  
+    1. If <var>layout shift</var> is greater than 0, create a new {{LayoutShift}} object |newEntry| and set its attributes as follows:
+        1. Set |newEntry|'s {{PerformanceEntry/name}} attribute to <code>"layout-shift"</code>.
+        1. Set |newEntry|'s {{PerformanceEntry/entryType}} attribute to <code>layoutShift</code>.
```

Comment:
Should `layoutShift` also be in quotes (as `"layout-shift"` is above)?

Reply:
yes done